### PR TITLE
Add Compass Less operation page

### PR DESCRIPTION
**Pull request by Hwurzburg**

### CODE REVIEW

File: common/source/docs/common-compassless.rst [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+===========================
+Operation without a Compass
+===========================
+
+Operation without a compass has been possible for Plane for a long time. However, recent additions of a compass/yaw source fallback mechanism in ArduPilot has opened the possibility of operations of  VTOL vehicles, such as Copters and QuadPlanes, without the use of a compass.
+
+On Plane, a steady GPS heading is used after takeoff to align the /attitude/velocity/position/heading estimation filters (EKFs) for yaw. This usually occurs a few seconds after takeoff on Planes without compass. However, this approach cannot be used for VTOL navigation, since it needs to be accurate very shortly after takeoffs with no GPS velocity or heading inputs.
+
+Compass-less VTOL operation has been made possible by the addition of a Gaussian Sum Filter (GSF) which takes IMU and GPS inputs to compute a probable heading,if a yaw source (like a compass) is unavailable, or unhealthy.
```
Comment (tridge) [inline common/source/docs/common-compassless.rst:11]: missing space after comma

File: common/source/docs/common-compassless.rst [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+.. note:: So far, the accuracy is quite acceptable, but this is still considered experimental and more experience and test flights by the community are required.
```
Comment (tridge) [inline common/source/docs/common-compassless.rst:13]: I think change "So far" to "In testing so far"

File: common/source/docs/common-compassless.rst [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+Setup
+=====
+
+Simply, set :ref:`COMPASS_ENABLE<COMPASS_ENABLE>` = 0. Fallback to the GSF will be automatic if normal default values are being used for EKF parameters.
+
+.. note:: Compass-less operation depends on good velocity reports from the GPS. Do not attempt with older generation GPS, use at least UBlox MN8 or equivalent performance GPS. Also, be sure GPS view of the sky is unobstructed throughout the flight. Otherwise, poor heading estimation will result.
```
Comment (tridge) [inline common/source/docs/common-compassless.rst:20]: change MN8 to M8

**Review by tridge — CHANGES_REQUESTED** [review]

File: common/source/docs/common-compassless.rst [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+
+#. Disable :ref:`ARMING_CHECK<ARMING_CHECK>` for Compass (keep the others!).
+#. Force arming from ground station.
+#. Before arming, but after GPS lock has been obtained and EKF origin has been set and is "using GPS", pick up the vehicle and walk around in a circle a few meters in diameter. This should allow the GSF to acquire yaw alignment and the message "EKF yaw alignment complete" would be sent to the ground station.
+
+.. warning:: If either of the first two methods are used, then the first movements of the vehicle in VTOL position controlled modes may be initially in the wrong direction, but should quickly stop once yaw alignment of the GSF has been obtained. Either fly the vehicle a bit in a non-position controller mode (like STABILIZE or QHOVER), or give the vehicle sufficient clearance for the unknown initial movement direction.
+
+Afterwards, flight and navigation should be acceptable.
+
+If any problems are experienced, please post logs either on the https://discuss.ardupilot.org or via a GitHub issue at https://github.com/ArduPilot/ardupilot/issues.
```
Comment (rmackay9) [inline common/source/docs/common-compassless.rst:35]: I don't think we should recommend users directly create issues, best to just stick with the discuss option.

File: common/source/docs/common-compassless.rst [hunk 5]
```diff
@@ -0,0 +1,37 @@
+.. _common-compassless:
+
+===========================
+Operation without a Compass
+===========================
+
+Operation without a compass has been possible for Plane for a long time. However, recent additions of a compass/yaw source fallback mechanism in ArduPilot has opened the possibility of operations of  VTOL vehicles, such as Copters and QuadPlanes, without the use of a compass.
```
Comment (rmackay9) [inline common/source/docs/common-compassless.rst:7]: I think "recent additions of a compass/yaw source fallback mechanism in ArduPilot" could be changed to "ArduPilot 4.1 (and higher) includes a Gaussian Sum Filter (GSF) which may allow Copter, Rovers and QuadPlanes to fly without a compass"

**Comment by rmackay9** [comment]
Maybe we should add a link to my video: https://www.youtube.com/watch?v=3xW9hj-lxNU.  We've got some other feedback above but after these are addressed I think we should merge this.  Perhaps we should also add a note near the top of the page that GSF is only available in ArduPilot 4.1 (and higher)

**Comment by Hwurzburg** [comment]
no need for notes on rev (I am against adding things that obsolete themselves over time, ie when firmware is beyond the note)....will add this to the Upcoming features page and it can be moved into mainstream wiki when it appears in a beta release...

File: common/source/docs/common-compassless.rst (hunk 4 above)
Comment (Hwurzburg) [reply common/source/docs/common-compassless.rst:35]: done

**Comment by Hwurzburg** [comment]
> Maybe we should add a link to my video: https://www.youtube.com/watch?v=3xW9hj-lxNU. We've got some other feedback above but after these are addressed I think we should merge this. Perhaps we should also add a note near the top of the page that GSF is only available in ArduPilot 4.1 (and higher)

added video, corrected, moved links to Upcoming so no version tag is required....merging...thanks!

**Comment by AndKe** [comment]
Please add information that gives answers to this:
https://discuss.ardupilot.org/t/compassless-flight-plus-gsf-more-questions-than-answers/63732

As-is, it is hard to fully understand if GSF is supposed to be an aid to enable compassless flight when prepared for it, or as handling of compass error after takeoff,  or a fallback for all stages of flight.

**Pull request merged**